pkg/certificate/manager_test.go

```diff
@@ -365,3 +364,47 @@ func TestHandleMRCEvent(t *testing.T) {
 		})
 	}
 }
+
+func TestSubscribeRotations(t *testing.T) {
+	assert := tassert.New(t)
+	cnPrefix1 := "fake-cert-cn1"
+	cnPrefix2 := "fake-cert-cn2"
+
+	cm := &Manager{
+		serviceCertValidityDuration: func() time.Duration { return time.Hour },
+		// The root certificate signing all newly issued certificates
+		signingIssuer:    &issuer{ID: "id1", Issuer: &fakeIssuer{id: "id1"}, CertificateAuthority: pem.RootCertificate("id1"), TrustDomain: "fake1.domain.com"},
+		validatingIssuer: &issuer{ID: "id1", Issuer: &fakeIssuer{id: "id1"}, CertificateAuthority: pem.RootCertificate("id1"), TrustDomain: "fake1.domain.com"},
+		pubsub:           pubsub.New(0),
+	}
+
+	cert, err := cm.IssueCertificate(cnPrefix1, Service)
+	assert.NoError(err)
+	assert.Equal("fake-cert-cn1.fake1.domain.com", cert.GetCommonName().String())
+
+	cert, err = cm.IssueCertificate(cnPrefix2, Service)
+	assert.NoError(err)
+	assert.Equal("fake-cert-cn2.fake1.domain.com", cert.GetCommonName().String())
+
+	ch1, unsub1 := cm.SubscribeRotations(cnPrefix1)
+	ch2, unsub2 := cm.SubscribeRotations(cnPrefix2)
```

Comment:
Should we subscribe here before issuing? Even though in this scenario we won't miss the rotation, would it be beneficial to be consistent about calling subscribe before issuing?

Reply:
ya i could see that, done now